src/models/candidature-process/candidature-process.service.ts

```diff
@@ -116,4 +114,40 @@ export class CandidatureProcessService {
     return embed;
   }
 
+  /**
+   * Emojis vote callback
+   * @param candidat The candidat for which the vote is made
+   */
+  candidatureVoteCallback(candidat: DeepPartial<UserEntity>){
+    return async (emoji:MessageReaction, user:User) => {
+     switch (emoji.emoji.name) {
+        case '👍':
+          await this.vote(user.id, candidat,true);
+          break;
+        case '👎':
+          await this.vote(user.id, candidat, false);
+          break;
+     }
+    }
+  }
+
+  /**
+   * Vote handler
+   * @parm discordID The discord id of the user
+   * @parm vote true if the vote is positive false if negative
+   */
+  async vote(discordID, candidat: DeepPartial<UserEntity>, vote: boolean){
+    const user = await this.userService.findOne(discordID);
+    if (user) {
```

Comment:
idem avec 
if(!user)

+  await this.botUtilityService.sendPrivateMessage similaire, on pourrait pas créer une var msg et le retourner à la fin ?

Reply:
je pense que tu à review une ancienne version je retrouve pas se bou de code